src/lxc/confile.c

```diff
@@ -291,21 +295,24 @@ static int set_config_net_type(const char *key, const char *value,
 	if (!netdev)
 		return -1;
 
-	if (!strcmp(value, "veth")) {
+	if (strcmp(value, "veth") == 0) {
```

Comment:
@brauner based on https://github.com/lxc/lxc/blob/master/CODING_STYLE.md#11-non-boolean-functions-that-behave-like-boolean-functions-must-explicitly-check-against-a-value I spotted a few existing issues here, as well as one new one I've added for ipvlan, so fixed them all here.